src/components/blocks/image.js

```diff
@@ -254,6 +255,10 @@ export default class ImageBlock extends React.Component {
   }
 
   uploadFile() {
+    if (this.config.upload_handler) {
+      return this.config.upload_handler(this.formatData().get('file'), this)
```

Comment:
@marekolszewski if you hit return here then the uploadCompleted will not be executed , right ?